packages/catalog-process/src/services/readModelService.ts

```diff
@@ -146,27 +151,73 @@ export function readModelServiceBuilder(
         ],
       };
 
+      const descriptorsFilter =
+        states.length === 0 && attributesIds.length === 0
+          ? {}
+          : {
+              "data.descriptors": {
+                $elemMatch: {
+                  ...ReadModelRepository.arrayToFilter(
+                    states,
+                    descriptorsStateFilter
+                  ),
+                  ...ReadModelRepository.arrayToFilter(
+                    attributesIds,
+                    descriptorsAttributesFilter
+                  ),
+                },
+              },
+            };
+
+      const visibilityFilter: ReadModelFilter<EService> = hasPermission(
+        [userRoles.ADMIN_ROLE, userRoles.API_ROLE],
+        authData
+      )
+        ? {
+            $nor: [
+              {
+                $and: [
+                  { "data.producerId": { $ne: authData.organizationId } },
+                  { "data.descriptors": { $size: 0 } },
+                ],
+              },
+              {
+                $and: [
+                  { "data.producerId": { $ne: authData.organizationId } },
+                  { "data.descriptors": { $size: 1 } },
+                  {
+                    "data.descriptors": {
+                      $elemMatch: { state: { $eq: descriptorState.draft } },
+                    },
+                  },
```

Comment:
is this enough?
```suggestion
                  { "data.descriptors.state": { $eq: descriptorState.draft } },
```

Reply:
Thanks! I didn't know about [this](https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/#single-query-condition).